internal/k8s/reconciler/utils.go

```diff
@@ -119,6 +120,61 @@ func routeAllowedForListenerNamespaces(ctx context.Context, gatewayNS string, al
 	return false, nil
 }
 
+// routeAllowedForHTTPBackend determines whether the route is allowed
+// for the backend either by being in the same namespace or by having
+// an applicable ReferencePolicy in the same namespace as the backend.
+//
+// TODO This func is currently called once for each backendRef on a route and results
+//   in fetching ReferencePolicies more than we technically have to in some cases
+func routeAllowedForHTTPBackend(ctx context.Context, route *gw.HTTPRoute, backendRef gw.HTTPBackendRef, c gatewayclient.Client) (bool, error) {
```

Comment:
Since it looks like there's nothing `HTTPRoute` specific here (only `GetNamespace()` and `GroupVersionKind()` called) I'm assuming you're going to modify this to be `route Route` instead of `route *gw.HTTPRoute` and `backendRef gw.BackendRef` instead of `backendRef gw.HTTPBackendRef` (`HTTPBackendRef` [embeds](https://github.com/kubernetes-sigs/gateway-api/blob/ef302d4b80d097f7b072a506adcf01b33194f27c/apis/v1alpha2/httproute_types.go#L841) `BackendRef`) so that we can have this be a generic `ReferencePolicy` implementation?

Reply:
Getting there with the next set of items on the checklist above. I usually work by starting with the specific thing and then generalizing